python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -317,6 +322,24 @@ def read_messages(
     def on_opened(self, opened_payload: PipesOpenedData) -> None:
         self.opened_payload = opened_payload
 
+    def update_params(
+        self,
+        message_reader_params: Optional[Mapping[str, Any]],
+        log_readers_params: Optional[Mapping[str, Mapping[str, Any]]],
+    ) -> None:
+        """Can be called manually to submit extra params (for example, only known after the external process has been launched).
+        Params for MessageReader should be written to state["params"]["message_reader"].
+        Params for LogReaders should be written to state["params"]["log_readers"].
+        Emits runtime_params_updated event to notify the message reader thread about new params being available.
+        """
+        if message_reader_params:
+            self.state["params"]["message_reader"].update(message_reader_params)
+
+        if log_readers_params:
+            self.state["params"]["log_readers"].update(log_readers_params)
+
+        self.runtime_params_updated.set()
```

Comment:
I'm now sure if we actually need this event, seems like checking for `self.target_is_readable` is enough